### DISCUSSION TITLE
CLI command *Compile* is missing

### DISCUSSION THREAD

**Discussion by rydlobeton** [superfaceai/.github#12] [Q&A] [answered]
Hello, 

after lastest update I cannot find Compile  command:

> Superface CLI utility

VERSION
  @superfaceai/cli/4.1.0 win32-x64 node-v18.12.1

USAGE
  $ superface [COMMAND]

COMMANDS
  execute  Run the created integration in superface directory. Commands `prepare`,  
           `new` and `map` must be run before this command. You can switch runner   
           language via `language` flag (`js` by default).
  login    Login to superface server
  logout   Logs out logged in user
  map      Creates a new (or updates an existing) Comlink Map that maps the use case
           to the selected API provider. After Map is available, the integration is 
           ready to be used by our WASM OneSDK. You should check security,
           integration parameters and input in the created files before execution.  
           The created Comlinks can be tested by running `superface execute` command
  new      Creates new Comlink Profile for your use case based on the selected API. 
           Comlink Profile defines the interface of the API integration. Use name of
           API provider as the first argument followed by the description of your   
           use case. You need to run `superface prepare` command before running this
           command.
  prepare  Learns API from the documentation and prepares the API metadata.
  whoami   Prints info about logged in user<

I've tried to reinstall @superfaceai package several times, but it didnt helped either.

And after going back to 3.0.1 am getting this new error while running code:

`AssertionError: $.name: must match pattern "^[a-z][_\-a-z]*$"
    at new AssertionError (\node_modules\@superfaceai\ast\dist\error.js:55:28)
    at assert (\node_modules\@superfaceai\ast\dist\validation.js:125:19)
    at Object.assertProviderJson (\node_modules\@superfaceai\ast\dist\interfaces\providerjson\providerjson.utils.js:45:5)
    at Object.<anonymous> (\node_modules\@superfaceai\one-sdk\dist\core\provider\resolve-provider-json.js:80:42)   
    at step (\node_modules\@superfaceai\one-sdk\dist\core\provider\resolve-provider-json.js:33:23)
    at Object.next (\node_modules\@superfaceai\one-sdk\dist\core\provider\resolve-provider-json.js:14:53)
    at fulfilled (\node_modules\@superfaceai\one-sdk\dist\core\provider\resolve-provider-json.js:5:58) {
  errors: [ [ 'must match pattern "^[a-z][_\\-a-z]*$"', [Array] ] ],
  data: {
    name: 'main_upgates_client3',
    services: [ [Object] ],
    defaultService: 'api',
    securitySchemes: [ [Object] ]
  }
}`

Any Idea how to fix this?

Programming on Windows 11, using LF convention.

Thanks in advance!

Rydlobeton

**Comment by kysely** [comment, marked as answer]
Hi @rydlobeton,

I apologize for the inconvenience — we deprecated commands related to [the previous OneSDK version (pre 3, for Node.js only)](https://superface.ai/docs/classic) and removed them in the latest release.

Downgrading CLI to the last major 3 version will support your workflow.

Regarding the second error, it seems like an issue with **the number in the provider name**. As per the regex presented in the error, the provider name must _start with a lowercase letter_, followed by either a _lowercase letter_, an underscore (`_`) or a dash (`-`).

Given that, the given provider name `main_upgates_client3` is invalid.

Hope this helps!
Radek

**Reply by rydlobeton** [reply]
Thanks for reply, i've never had an issue with given name in previous releases, where can I read all the new rules?

Best regards
Rydlobeton

**Reply by zdne** [reply]
@rydlobeton I am afraid this was always in the spec, it was just poorly implemented and as such it allowed (incorrectly) the numbers. 

The spec for it is in https://github.com/superfaceai/spec/blob/main/spec/latest/provider-spec.yaml#L6 and https://superface.ai/docs/comlink/reference/provider#json-schema

**Reply by zdne** [reply]
@rydlobeton if you would like, I'd be happy to jump on a call and discuss the future direction with superface and comlinks!

**Reply by rydlobeton** [reply]
Gladly, 

I'd been in contact with your fellow collegue, but lost contact with him.

Superface helped me to implement and achieve my goals even before id leanrt NodeJS.

I gladly share my experience from using it on lower level and learn something new in return. Let me know how to contact you.

Best regards

Rydlobeton
